### DISCUSSION TITLE
classes/class_polygon2d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#262] [User-contributed notes]
# classes/class_polygon2d

Inherits: Node2D< CanvasItem< Node< Object A 2D polygon. Description: A Polygon2D is defined by a set of points. Each point is connected to the next, with the final point being connected to the fir...

https://docs.godotengine.org/en/stable/classes/class_polygon2d.html

**Comment by soradil** [comment]
Sometimes, a "smudging" issue arises when mapping UV points from a Texture image onto a Polygon2D. It seems to have something to do with a clamping problem within Godot that occur when the image touches the boundary. 
To fix this, in any image editing app like photoshop, inkscape or gimp, add a margin around the content of the image of 1 transparent pixel.

**Comment by gostke** [comment]
The Polygon2D's vertex positions are local, not global, so keep in mind to transform them if you want to get their real world coordinates.
